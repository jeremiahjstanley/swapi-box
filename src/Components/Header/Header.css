
.header {
  background-color: #222;
  height: 15vh;
  margin: auto
}

.button {
  background: linear-gradient(#fff, #222, #222);
  border-radius: 50px;
  border: .1rem solid #fff;
  color: #fff;
  font-family: Helvetica;
  font-size: 1.1rem;
  font-weight: 600;
  margin: 1rem;
  padding: 1rem;
  text-align: center;
  text-transform: uppercase;
}

span {
  padding-left: .5rem;
  color: #00aeef;
}

.active {
  border: .1rem solid #00aeef; 
  color: #00aeef; 
}