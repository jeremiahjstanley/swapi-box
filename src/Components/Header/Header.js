import React from 'react';
import PropTypes from 'prop-types'
import './Header.css'

const Header = ({buttons, displayCards, favorites}) => {

  const handleClick = (button) => {
    buttons.map(button => button.active = false)
    button.active = true;
    displayCards(button.name);
  }

  const determineButton = (button) => {
    if (button.name === 'favorites') {
      return (        
        <p>
         {favorites.length}
        </p>
      )
    }
  }

  const clickButtons = buttons.map((button, index) => {
    return (
      <button
        className={button.active ? 'button active' : 'button'}
        key={index}
        onClick={() => {handleClick(button)}}
      >
        {button.name}
        {determineButton(button)}
      </button>
    )
  })

return (
  <div>
    {clickButtons}
  </div>
  )
}

Header.propTypes = {
  buttons: PropTypes.array.isRequired,
  displayCards: PropTypes.func.isRequired,
  favorites: PropTypes.array.isRequired
}


export default Header;