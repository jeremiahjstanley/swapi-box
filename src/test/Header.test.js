import React from 'react'
import { shallow } from 'enzyme'
import Header from '../Components/Header/Header'

describe('Header Tests', () => {
  let wrapper;
  let mockButtons;
  let mockDisplayCards;
  let mockFavorites;

  beforeEach(() => {
    mockButtons = [{name: 'people', active: false},
                    {name: 'planets', active: false},
                    {name: 'vehicles', active: false},
                    {name: 'favorites', active: false}]
    mockFavorites = []
    mockDisplayCards = jest.fn()
    wrapper = shallow(<Header 
                        buttons={mockButtons} 
                        displayCards={mockDisplayCards}
                        favorites={mockFavorites}
                      />)
  })

  it('should match the snapshot',() => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render four buttons', () => {
    const actual = wrapper.find('.button').length
    const expected = 4
    expect(expected).toEqual(actual)
  })

  it('buttons should call displayCards when clicked', () => {
  })


})