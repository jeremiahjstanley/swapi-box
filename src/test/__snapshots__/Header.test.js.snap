// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    buttons={
        Array [
            Object {
              "active": false,
              "name": "people",
            },
            Object {
              "active": false,
              "name": "planets",
            },
            Object {
              "active": false,
              "name": "vehicles",
            },
            Object {
              "active": false,
              "name": "favorites",
            },
          ]
    }
    displayCards={[Function]}
    favorites={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="button"
          onClick={[Function]}
>
          people
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          planets
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          vehicles
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          favorites
          <p />
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "people",
            undefined,
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "people",
          undefined,
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "planets",
            undefined,
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "planets",
          undefined,
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "vehicles",
            undefined,
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "vehicles",
          undefined,
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "favorites",
            <p />,
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "favorites",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "p",
          },
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="button"
            onClick={[Function]}
>
            people
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            planets
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            vehicles
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            favorites
            <p />
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "people",
              undefined,
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "people",
            undefined,
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "planets",
              undefined,
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "planets",
            undefined,
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "vehicles",
              undefined,
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "vehicles",
            undefined,
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "favorites",
              <p />,
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "favorites",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "p",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
